cli/dcoscli/job/main.py

```diff
@@ -0,0 +1,857 @@
+
+
+def main(argv):
+    try:
+        return _main(argv)
+    except DCOSException as e:
+        emitter.publish(e)
+        return 1
+
+
+@decorate_docopt_usage
+def _main(argv):
+    args = docopt.docopt(
+        default_doc("job"),
+        argv=argv,
+        version='dcos-job version {}'.format(dcoscli.version))
+
+    return cmds.execute(_cmds(), args)
+
+
+def _cmds():
+    """
+    :returns: all the supported commands
+    :rtype: dcos.cmds.Command
+    """
+
+    return [
+
+        cmds.Command(
+            hierarchy=['job', 'run'],
+            arg_keys=['<job-id>'],
+            function=_run),
+
+        cmds.Command(
+            hierarchy=['job', 'kill'],
+            arg_keys=['<job-id>', '<run-id>', '--all'],
+            function=_kill),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'add'],
+            arg_keys=['<job-id>', '<schedule-file>'],
+            function=_add_schedule),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'update'],
+            arg_keys=['<job-id>', '<schedule-file>'],
+            function=_update_schedules),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'show'],
+            arg_keys=['<job-id>', '--json'],
+            function=_show_schedule),
+
+        cmds.Command(
+            hierarchy=['job', 'show', 'runs'],
+            arg_keys=['<job-id>', '<run-id>', '--json', '--q'],
+            function=_show_runs),
+
+        cmds.Command(
+            hierarchy=['job', 'schedule', 'remove'],
+            arg_keys=['<job-id>', '<schedule-id>'],
+            function=_remove_schedule),
+
+        cmds.Command(
+            hierarchy=['job', 'list'],
+            arg_keys=['--json'],
+            function=_list),
+
+        cmds.Command(
+            hierarchy=['job', 'history'],
+            arg_keys=['<job-id>', '--json', '--show-failures'],
+            function=_history),
+
+        cmds.Command(
+            hierarchy=['job', 'remove'],
+            arg_keys=['<job-id>', '--stopCurrentJobRuns'],
+            function=_remove),
+
+        cmds.Command(
+            hierarchy=['job', 'add'],
+            arg_keys=['<job-file>'],
+            function=_add_job),
+
+        cmds.Command(
+            hierarchy=['job', 'update'],
+            arg_keys=['<job-file>'],
+            function=_update_job),
+
+        cmds.Command(
+            hierarchy=['job', 'show'],
+            arg_keys=['<job-id>'],
+            function=_show),
+
+        cmds.Command(
+            hierarchy=['job'],
+            arg_keys=['--config-schema', '--info'],
+            function=_job)
+    ]
+
+
+def _job(config_schema=False, info=False):
+    """
+    :param config_schema: Whether to output the config schema
+    :type config_schema: boolean
+    :param info: Whether to output a description of this subcommand
+    :type info: boolean
+    :returns: process return code
+    :rtype: int
+    """
+
```

Comment:
@kensipe None of these commands actually check the `metronome` capability. We need to add that before we can merge: see https://github.com/dcos/dcos-cli/blob/master/cli/dcoscli/node/main.py#L121